clusters/resource_cluster.go

```diff
@@ -410,8 +410,19 @@ func resourceClusterUpdate(ctx context.Context, d *schema.ResourceData, c *commo
 			})
 		} else {
 			var editCluster compute.EditCluster
-			editCluster.ClusterId = clusterId
 			common.DataToStructPointer(d, clusterSchema, &editCluster)
+
+			// We should have the an interface for all cluster structures so that we can use common modification among different structs. This should be done in OpenAPI spec.
```

Comment:
We could make the functions generic and get the fields using reflection but it would become more difficult to maintain.